hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/hudi/SparkAdapter.scala

```diff
@@ -232,7 +231,13 @@ trait SparkAdapter extends Serializable {
   def createParquetFileReader(vectorized: Boolean,
                               sqlConf: SQLConf,
                               options: Map[String, String],
-                              hadoopConf: Configuration): SparkParquetReader
+                              hadoopConf: Configuration): SparkColumnarFileReader
+
+  def createOrcFileReader(vectorized: Boolean,
+                          sqlConf: SQLConf,
+                          options: Map[String, String],
+                          hadoopConf: Configuration,
+                          dataSchema: StructType): SparkColumnarFileReader
```

Comment:
Why do we need to pass the data schema here? Can we avoid this?

Reply:
It's used in the code:
```
val resultedColPruneInfo = OrcUtils.requestedColumnIds(
        isCaseSensitive, dataSchema, requiredSchema, orcSchema, conf)
```
I think that we may be able to use the file schema as the data schema though